fgpyo/util/metric.py

```diff
@@ -466,3 +477,103 @@ def asdict(metric: Metric) -> Dict[str, Any]:
             "The provided metric is not an instance of a `dataclass` or `attr.s`-decorated Metric "
             f"class: {metric.__class__}"
         )
+
+
+def _get_fieldnames(metric_class: Type[Metric]) -> List[str]:
+    """
+    Get the fieldnames of the specified metric class.
+
+    Args:
+        metric_class: A Metric class.
+
+    Returns:
+        A list of fieldnames.
+
+    Raises:
+        TypeError: If the given class is not a Metric.
```

Comment:
The `TypeError` is raised by `_assert_is_metric_class()`, and I should remove it from the doc. (I prefer to only document exceptions directly raised by the function)

I will remove the `Raises` block because the assertion is unreachable.